[WIP] Implement perceptual indexing and search

G10h4ck opened the pull request:
This pull request is a work in progress to implement perceptual file indexing and search capabilities in libretroshare. Once complete it should be possible to run searches over the RetroShare network to find similar files (at least images, audio and videos) like big G does for images. As an example with perceptual search it will be easy to find a version with better quality of an audio or an image having a worse one, as the two versions should produce very "similar" perceptual hashes.

G10h4ck committed:
Implement proper GXS message deletion notification

GxsForums propagate message deletion notification as RsEvent

G10h4ck committed:
Notify one deletion per event as Cyril suggested

G10h4ck committed:
Use unannotated pointer as per Cyril taste

G10h4ck committed:
First working prototype of GXS service search

Channels are now able to take advantage of the new deep search
  goodies

G10h4ck committed:
First working prototype of GXS service search

Channels are now able to take advantage of the new deep search
  goodies

Rebase on top of master 2021/10/19

G10h4ck committed:
Implement deep indexing and search for forums

RsGxsNetTunnelService::receiveSearchRequest handle no results case
  properly
RsNxsObserver::handleDistantSearchRequest improve method behaviour
  documentation
RsTurtleClientService Improve documentation

G10h4ck committed:
Implement deep indexing and search for forums

RsGxsNetTunnelService::receiveSearchRequest handle no results case
  properly
RsNxsObserver::handleDistantSearchRequest improve method behaviour
  documentation
RsTurtleClientService Improve documentation

G10h4ck committed:
Epurate this branch from channels deep search changes

G10h4ck committed:
Implement pull request mechanism in RsGxsNetService

This could be used to request the online peers to pull updates from us
  ASAP, as an exaple when a group is created a pull request can be
  emitted too so the online peers pull the groups from us ASAP instead
  of waiting for the usual 60 seconds. A mechanism like this is
  especially useful on mobile phones where the internet connection is
  usually turned on only in a few moments (as an example while the user
  is interacting with the app).
Cleanup a few old corners in the code keeping retro-compatibility and
  make the code more welcoming to new developers.
Put a bunch of dead code under #ifdef.

G10h4ck committed:
Implement RsNxsSerialiser::create_item for RsNxsPullRequestItem

G10h4ck committed:
Request pull from peers when GXS group is created

G10h4ck committed:
Fix includes in rsnxs.h

G10h4ck committed:
Request pull from peers when GXS message is created

G10h4ck committed:
RsGxsNetService trigger pulling on group subscribe

G10h4ck committed:
Forums expose API to request syncronization

G10h4ck committed:
Fix Android compilation

G10h4ck committed:
RsGxsForumsDistantSearchEvent fix event type

G10h4ck committed:
Add libraries for perceptual search also for Android

On Android it has been expecially long to fiddle with all build systems
  libriaries bugs

G10h4ck committed:
Add image perceptual indexing prototype

G10h4ck committed:
Partial integration of perceptual search in file turtle search

G10h4ck committed:
Integrate perceptual index with turtle search

felisucoibi commented:
is nice but is something nobody asked for and lot of things that still needs to be done like a working web RS

G10h4ck commented:
> is nice but is something nobody asked for and lot of things that still needs to be done like a working web RS

None that you know, but someone that you don't may even have donated support for this cause

felisucoibi commented:
good to know this is the way

G10h4ck committed:
Properly avoid private forum leak in deep search

mGroupFlags & GXS_SERV::FLAG_PRIVACY_PUBLIC is always true for forums
  even if they are circle restricted, use circle flags to check if it
  is really public

G10h4ck committed:
Merge branch 'gxs_fast_pull_request' into elrepo

G10h4ck committed:
Merge branch 'notify_hashing_file_completed' into elrepo

G10h4ck committed:
Perceptual search made optional and disabled by default
